src/map/lua/lua_baseentity.cpp

```diff
@@ -14507,7 +14507,20 @@ uint16 CLuaBaseEntity::getWeaponDmg()
         return 0;
     }
 
-    return static_cast<CBattleEntity*>(m_PBaseEntity)->GetMainWeaponDmg();
+    uint16 weaponDamage = 0;
+
+    if (m_PBaseEntity->objtype == TYPE_MOB ||
+        m_PBaseEntity->objtype == TYPE_PET)
+    {
```

Comment:
What about trusts and such? Wouldnt it be easier to just check for PC?

Reply:
I was thinking that most things that aren't mobs would use the player damage formula, if I'm mistaken and its the other way around I can reverse the check to be only for PC

Reply:
I feel like this is a problem for another day, I don't think anyone actually knows what the base damage of a trust is. there's no way its calculated based on level alone (like mobs/pets), it's probably completely arbitrary for each and every one

I would throw in a TODO and kick that can down the road

Reply:
Got a TODO for the trusts and fellows on there now